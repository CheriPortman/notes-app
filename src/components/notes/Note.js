import React from 'react';
import PropTypes from 'prop-types';

import { deleteNote } from '../../services/notesApi';
import { Button, PT } from '../styledComponents';

function Note({ note }) {
  return (
    <section>
      <PT>{note.title}</PT>
      <p>{note.body}</p>
      <Button onClick={() => deleteNote(note._id)}>Delete This Note</Button>
    </section>
  );
}
// deleteNote.bind(null, note._id)
Note.propTypes = {
  note: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired
  }).isRequired
};

export default Note;
