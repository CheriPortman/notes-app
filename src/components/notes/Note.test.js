import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Note from './Note';

describe('Note component', () => {
  it('renders the Note app', () => {
    const wrapper = shallow(<Note note={{ title: 'note', text: 'note' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
