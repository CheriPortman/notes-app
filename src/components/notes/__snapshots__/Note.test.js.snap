// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders the Note app 1`] = `
<section>
  <styled.p>
    note
  </styled.p>
  <p />
  <styled.button
    onClick={[Function]}
  >
    Delete This Note
  </styled.button>
</section>
`;
